agent/structs/structs.go

```diff
@@ -2438,6 +2475,15 @@ func (r *KeyringResponses) New() interface{} {
 	return new(KeyringResponses)
 }
 
+// String converts message type int to string
+func (m MessageType) String() string {
+	s, ok := requestTypeStrings[m]
+	if ok {
+		return s
+	}
+	return "Unknown(" + strconv.Itoa(int(m)) + ")"
```

Comment:
I like that this returns the unknowns vs crashing the way the old snapshot tool did